pyro/infer/mcmc/hmc.py

```diff
@@ -18,17 +20,24 @@ class HMC(TraceKernel):
     Simple Hamiltonian Monte Carlo kernel, where ``step_size`` and ``num_steps``
     need to be explicitly specified by the user.
 
-    **Reference**
-    "MCMC Using Hamiltonian Dynamics", R Neal. `pdf <https://arxiv.org/pdf/1206.1901.pdf>`_
+    References
 
+    [1] `MCMC Using Hamiltonian Dynamics`,
+    Radford M. Neal
 
-    :param model: python callable containing pyro primitives.
-    :param float step_size: determines the size of a single step taken by the
+    :param model: Python callable containing pyro primitives.
+    :param float step_size: Determines the size of a single step taken by the
         verlet integrator while computing the trajectory using Hamiltonian
-        dynamics.
+        dynamics. If not specified, it will be set to 1.
+    :param float trajectory_length: Length of a MCMC trajectory. If not
```

Comment:
It will also be nice to avoid interactions between `trajectory_length`, `num_steps` and `step_size`. If we need to be able to specify `trajectory_length` for adapting the `step_size`, can we just remove `step_size` as a parameter. So, our parameters are just `HMC(model, num_steps=None, trajectory_length)`. Then if `num_steps` is not specified we just use the dual averaging version.